code/iaas/allocator/src/main/java/io/cattle/platform/allocator/service/AllocatorService.java

```diff
@@ -0,0 +1,18 @@
+package io.cattle.platform.allocator.service;
+
+import io.cattle.platform.allocator.constraint.Constraint;
+
+import java.util.List;
+import java.util.Map;
+
+public interface AllocatorService {
+
+    public List<Long> getHostsForGlobalService(Long accountId, Map<String, String> labels);
```

Comment:
well, just the HostAffinity label requirement.  What about this name?
getHostsFilteredByHostAffinityConstraint()
In the future, I might refactor some more if the need arises